src/kbc_modules/cc_kbc/crypto/mod.rs

```diff
@@ -17,7 +17,20 @@ const RSA_ALGORITHM: &str = "RSA1_5";
 const RSA_PUBKEY_LENGTH: usize = 2048;
 const NEW_PADDING: fn() -> PaddingScheme = PaddingScheme::new_pkcs1v15_encrypt;
 
-const AES_GCM_256_ALGORITHM: &str = "A256GCM";
+pub const AES_GCM_256_ALGORITHM: &str = "A256GCM";
+
+#[derive(Serialize, Deserialize)]
+pub struct AnnotationPacket {
+    // Key Resource URL
+    // e.g: http://127.0.0.1:8080/kbs/v0/resource/test_repo/key/id_1
+    pub key_url: String,
```

Comment:
I have corrected this. The `kid` here should actually be a new unified format proposed in this issue: https://github.com/confidential-containers/documentation/issues/85. This format will be used in CoCo to index all resources that need to be obtained from AA. The `kid` here is consistent with it. The format is:

```
cc_kbc://127.0.0.1:8080/test_repo/key/id_1
```

Reply:
Hi @dubek , I'm going to bring the logic of parsing resource URL out of logic of KBCs. For example, 
- `cc_kbc://127.0.0.1:8080/test_repo/key/id_1`. AA will dispatch this url to `cckbc`, and let `cckbc` determine how to process `127.0.0.1:8080/test_repo/key/id_1`. In this scenario, it will convert this into a http link `https://127.0.0.1:8080//kbs/v0/resources/key/id_1` and make a request in kbs protocol
- `offlinefskbc://key/1`. AA will dispatch this url to `offline fs kbc`, and let `offline fs kbc` determine how to process `key/1`. Based on the mechinism of offline kbc, it will look up the resource.json to directly return.

The PR is under development https://github.com/confidential-containers/attestation-agent/pull/117
The mechinism is described in https://github.com/confidential-containers/documentation/issues/85